src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -272,16 +272,18 @@ private async Task HandleMessageAsync(Message message, CancellationToken cancell
 
             try
             {
-                var operationId = DetermineOperationId(message.CorrelationId);
-                var correlationInfo = new MessageCorrelationInfo(message.GetTransactionId(), operationId);
-
-                var messageContext = new AzureServiceBusMessageContext(message.MessageId, message.SystemProperties,
-                    message.UserProperties);
+                if (String.IsNullOrEmpty(message.CorrelationId))
+                {
+                    Logger.LogInformation("Operation ID will be generated as no ID was found on the message");
```

Comment:
The pump doesn't have to care about that, he just knows nothing is there, no?

```suggestion
                    Logger.LogInformation("No operation ID was found on the message");
```